course_catalog/etl/video.py

```diff
@@ -0,0 +1,66 @@
+"""video catalog ETL"""
+
+from django.conf import settings
+from googleapiclient.discovery import build
+from django.conf import settings
+
+class YouTubeApi:
+    """
+    Class interface to YouTube API calls
+    """
+
+    client = None
+
+    def __init__(self):
+        api_service_name = "youtube"
+        api_version = "v3"
+        developer_key = settings.YT_DEVELOPER_KEY
+
+        self.client = build(api_service_name, api_version, developerKey=developer_key)
+
+    def get_videos_for_channel_in_batches(self, channel_id, offered_by):
+        page_token = ""
+
+        while True:
+            request = self.client.search().list(
+                part="snippet",
+                channelId=channel_id,
+                maxResults=50,
+                pageToken=page_token
+            )
+
+            response = request.execute()
+            response["offered_by"] = offered_by
+
+         
+            for video in response['items']:
+                try:
+                    video_id = video['id']['videoId']
+                    video['url'] = 'https://www.youtube.com/watch?v=' + video_id
+                    video['snippet']['full_description'] = self.get_full_description(video_id)
+                except KeyError:
+                    print(video)
```

Comment:
This code is transforming the response, it's probably more appropriate for the work we'll do in #2304. I can see that you're doing this so that `offered_by` gets propagated. Fortunately the interface between `extract` and `transform` isn't a strict one so that can vary implementation-to-implementation.

I think what we want to do here is have `extract` yield `tuple`s like `(offered_by, item_data)` and then `transform will take a list of those. So your code here would look something like:

```python
for item in response["items"]:
    yield (offered_by, item)
```

A [named tuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) might also be good here since it provides a little more structure to the interface between `extract` and `transform`.

Reply:
Make sense! I would like to keep the line adding `full_description`, which requires an additional api call though. Does that make sense?

Reply:
Hmm, can we add that as a third item in the tuple instead?

Reply:
that would work. The list of item descriptions would be returned ordered the same way as item is